target/linux/ath79/dts/ar7100.dtsi

```diff
@@ -108,12 +108,18 @@
 				interrupt-parent = <&cpuintc>;
 				interrupts = <2>;
 
-				interrupt-controller;
 				#interrupt-cells = <1>;
-
-				interrupt-map-mask = <0 0 0 1>;
-				interrupt-map = <0 0 0 0 &pcie0 0>;
+				interrupt-map-mask = <0xf800 0 0 0>;
+				interrupt-map = <0x8800 0 0 0 &pci0_intc 1
+					0x9000 0 0 0 &pci0_intc 2
+					0x9800 0 0 0 &pci0_intc 3>;
```

Comment:
Maybe do the map the same. It doesn't really matter since these numbers can be any in the range, but just in case someone used them...
```
interrupt-map = <0x8800 0 0 0 &pci0_intc 0
￼ 			   0x9000 0 0 0 &pci0_intc 1
￼                           0x9800 0 0 0 &pci0_intc 2>;
```

Reply:
If you mean the hwirq number...The original map is wrong according to the driver. If we change these here we need more modifications to the driver code...

Reply:
The last number is not 1-4. It is the 3rd field that is INT1-4.
Hold on, I think we can do it a bit different way. I need to check one idea.

Reply:
I don't quite get the idea of changing hwirq number here. There are only 4 sub-interrupt for PCI interrupt and the hwirq here is assigned as the 1st/2nd/3rd/4th interrupt.
The 2nd and 3rd field should always be 0 because of the interrupt mask.
If you want to say the 4th field... Using interrupt-mask=<0xf800 0 0 7> and map it like this:
```
0x8800 0 0 0 &pci0_intc 1
0x8800 0 0 0 &pci0_intc 1
0x8800 0 0 0 &pci0_intc 1
0x8800 0 0 0 &pci0_intc 1
```
is identical to my map here.

Reply:
This makes no sense to add 4 of those. It could make sense like this

0x8800 0 0 1 &pci0_intc 0
0x8800 0 0 2 &pci0_intc 1
0x8800 0 0 3 &pci0_intc 2
0x8800 0 0 4 &pci0_intc 3

0x8800 0 0 1 &pci0_intc 3
0x8800 0 0 2 &pci0_intc 2
0x8800 0 0 3 &pci0_intc 0
0x8800 0 0 4 &pci0_intc 1

But I don't think we really nee this. But what confuses me that you use the hwirq to check the bit.